### PR TITLE
feat: file is saved as a object containing more data instead of just id

### LINKED ISSUE #614: Handling of files in the `content` field of `PbPage` model
## This is:
- Idea

## Actual Behavior

Before we start off, let's give some background.

## Background
In PageBuilder, we have `PpPage` model and in it, the `content` field.

The `content` field may store images. The following is an excerpt in
which we can see how a file is saved in the database.

```
(...)
    {
      "id": "Of2JQrobK",
      "data": {
        "image": {
          "file": "5db3410646e6da8d61cc1393"
        }
      },
      "elements": [],
      "path": "0.0.0.0.2",
      "type": "image"
    }
(...)
```

This value of `file` key here is a result of several steps:
1. In the Page Builder, user picks an image from File Manager. Note that the File Manager doesn't return the file's ID, but a full data object: `{ src, size, id, type, key }`. So once the client sends the revision update mutation, this whole object is sent.

2. On the backend, we have a set of plugins that completely traverse received `content`, and do the replacing upon saving to the database. So instead of the full received object, only `id` is saved, like in the first example shown above.

3. Again, when fetching data from the database, we have the opposite process, in which a set of plugins convert these IDs into `{ src, id }` objects, so that the frontend again has the initial structure od data. Note that only `src` and `id` values are returned, no need really for other keys. But it can be added if needed.

## The problem
The problem is currently happening in step 3. Currently, when we want to replace the ID with the object, we need to load the file, and we do that by directly accessing the database connection, and doing a query on `File` collection, which is not an ideal approach.

The second approach that was tried here was doing a GQL query on `api-files` service, which is much better in terms of microservices approach. But this time, the problem is that getting/listing pages becomes slow since each page has to do an HTTP request (1 page = 1 HTTP request that fetched multiple files). We have an even worse case when you do a `listPages`, in which you might query for 10 or more pages. A lot of HTTP requests and a lot of latency is being added.

So is there a third way that we could solve this?

## Possible Solution
To solve the above-explained problem, we could save files a bit differently.

Instead of saving the file `id` to the database, we would save file `key`. This is good because field `key` never changes and since we have it, we can easily construct the full URL to the file upon receiving the page data from the database.

We would only need to fetch the prefix for all files, which is stored in `api-files` service's settings, and which can be fetched via GQL. This would need to be fetched only once so it's not super intensive on the backend.

So to summarize, this would free us from making GQL requests (fetching all files by their IDs), since we would be able to construct full paths directly in code.

### Problem - File Manager File-Replacement Feature 
Question: do we want a file-replacement feature in the File Manager? In other words, can a user have a single image selected in multiple pages, and then go to the File Manager and choose that same image and replace it with another one? So basically `id` of the file stays the same, and the `key` to which it points changes.

If so, the above solution won't work, since it proposes saving file `key` instead of `id`. And changing the key of a single file via File Manager would not apply changes to the `content` field in pages.

### PR DESCRIPTION
**Pull request by brunozoric**
## Related Issue
Closes #614 #784 

## Status
- [x] correct storing/fetching the data in the existing modifiers and use `SettingsManager` to get the `srcPrefix`, always once
- [ ] adapt installation process to the newest changes
- [x] **[moved to #1250]** host installation files on a public S3  bucket

## Your solution
We are now saving file object that contains __type, id, name, size, type and meta object (height, width, aspectRatio, private).
When fetching, there is only one extra call to get srcPrefix so we can create src of the image.

## How Has This Been Tested?
Project was built from the start - yarn install -> setup-repo -> deploy -> save page, save settings general tab, save settings social tab

### CODE REVIEW

File: packages/api-page-builder/src/plugins/pageContentModifiers.ts [hunk 1]
```diff
@@ -14,41 +36,31 @@ export default [
         type: "pb-page-element-modifier",
         elementType: "*",
         getStorageValue({ element }) {
-            const file = get(element, "data.settings.background.image.file.id");
-            if (file) {
-                set(element, "data.settings.background.image.file", file);
-            }
+            const file = element?.data?.settings?.background?.image?.file;
+            element.data.settings.background.image.file = createFileForDatabase(file);
```
Comment (Pavel910) [inline packages/api-page-builder/src/plugins/pageContentModifiers.ts:40]: @brunozoric line 39 uses optional chaining, but line 40 assigns directly to the deeply nested key. Maybe we should use `set` to assign file value?

**Comment by doitadrian** [comment]
P.S. Don't forget to test the PB installation. 🙂  @brunozoric

**Comment by doitadrian** [comment]
BTW I see the installation isn't mentioned here, but I was sure I did mention it.

So I went searching, and this is basically a duplicate issue, but this one mentions the installation 🙂 

https://github.com/webiny/webiny-js/issues/784

**Comment by doitadrian** [comment]
@brunozoric @Pavel910 

Once we're done with the basic setup, we should also finally move the installation files into our public S3 bucket and have our installation process pull files from it. We don't need to bother users with uploading the ZIP in every new project.

I think we can just create a simple S3 bucket in our main AWS account, and host the files there. I remember @Pavel910 mentioning having a new folder for every version of Webiny, but at this point, since these files have NEVER changed, I think we can just drag and drop the ZIP, and get the URL.

**Comment by Pavel910** [comment]
@doitadrian we'll do that as the very last step before v5 release. We also need to migrate to new Slate, which will change the structure of data, and we'll need to update installation files.

**Comment by doitadrian** [comment]
Excellent @Pavel910, I'll write this as a new issue then.

**Pull request closed without merging**